python/datasources/vera_incarceration_county.py

```diff
@@ -251,93 +215,135 @@ def write_to_bq(self, dataset, gcs_bucket, **attrs):
+
+        # make partial breakdown for total_confined_children
+        partial_children_df = generate_partial_breakdown(
+            df.copy(), demo_type, JAIL, CHILDREN)
+
+        # merge in the column with confined children
+        breakdown_df = pd.merge(breakdown_df, partial_children_df, how="left", on=[
+            std_col.TIME_PERIOD_COL, *GEO_COLS_TO_STANDARD.values(), demo_col])
 
         # round 100k values
-        breakdown_df[RATE_COL_MAP[data_type]
-                     ] = breakdown_df[RATE_COL_MAP[data_type]].dropna().round()
+        for data_type in [PRISON, JAIL]:
+            breakdown_df[PER_100K_COL_MAP[data_type]
+                         ] = breakdown_df[PER_100K_COL_MAP[data_type]].dropna().round()
 
         breakdown_df[std_col.STATE_FIPS_COL] = breakdown_df[std_col.COUNTY_FIPS_COL].astype(
             str).str[:2]
 
+        # calculate pct_share cols for PRISON and POP (jail done already pre-melt)
         breakdown_df = generate_pct_share_col_without_unknowns(
             breakdown_df,
-            DATA_TYPE_TO_COL_MAP[data_type],
+            {**DATA_TYPE_TO_COL_MAP[PRISON],
+                POP: PCT_SHARE_COL_MAP[POP]},
             demo_col,
             all_val)
 
-        breakdown_df = generate_pct_share_col_without_unknowns(
-            breakdown_df,
-            {POP: PCT_SHARE_COL_MAP[POP]},
-            demo_col,
-            all_val)
-
-        # add a column with the confined children
-        breakdown_df = pd.merge(breakdown_df, df_children, how="left", on=[
-            *GEO_COLS_TO_STANDARD.values(), demo_col])
-
-        cols_to_drop = [std_col.POPULATION_COL,
-                        std_col.STATE_FIPS_COL, RAW_COL_MAP[data_type]]
-
-        breakdown_df = breakdown_df.drop(
-            columns=cols_to_drop)
-
+        # add relative inequity cols for jail and prison
+        for data_type in [PRISON, JAIL]:
+            breakdown_df = generate_pct_rel_inequity_col(
+                breakdown_df, PCT_SHARE_COL_MAP[data_type], PCT_SHARE_COL_MAP[POP], PCT_REL_INEQUITY_COL_MAP[data_type])
+            breakdown_df = zero_out_pct_rel_inequity(
+                breakdown_df,
+                "county",
+                cast(SEX_RACE_AGE_TYPE, demo_short),
+                {
+                    PER_100K_COL_MAP[data_type]: PCT_REL_INEQUITY_COL_MAP[data_type]
+                }
+            )
+
+        needed_cols = [std_col.TIME_PERIOD_COL,
+                       *GEO_COLS_TO_STANDARD.values(),
+                       demo_type,
+                       *PER_100K_COL_MAP.values(),
+                       *PCT_SHARE_COL_MAP.values(),
+                       *PCT_REL_INEQUITY_COL_MAP.values(),
+                       *RAW_COL_MAP.values(),
+                       POP,
+                       CHILDREN
+                       ]
+
+        # by_race gets race and race_id cols
         if demo_type == std_col.RACE_OR_HISPANIC_COL:
             std_col.add_race_columns_from_category_id(breakdown_df)
```

Comment:
Probably need to add `race_and_ethnicity` col here too?

Reply:
updated the comment; is that what you mean ?

Reply:
yeah